lambda/translcore.ml

```diff
@@ -714,7 +718,12 @@ and transl_apply ~scopes
         (* Evaluate the remaining arguments;
            if we already passed here this is a no-op. *)
         let l =
-          List.map (fun (arg, opt) -> Option.map (protect "arg") arg, opt) l
```

Comment:
I wonder if it would make sense to introduce `Typedtree.map_apply_arg` to replace `Option.map`, its absence creates some less-necessary code churn in this module (4 occurrences of `Option.map` have been inlined manually).